examples/asr/jasper.py

```diff
@@ -275,11 +276,14 @@ def main():
     neural_factory.train(
         tensors_to_optimize=[train_loss],
         callbacks=callbacks,
-        lr_policy=SquareAnnealing(args.num_epochs * steps_per_epoch,
-                                  warmup_steps=args.warmup_steps),
+        lr_policy=CosineAnnealing(
+            args.max_steps if args.max_steps is not None else
+            args.num_epochs * steps_per_epoch,
+            warmup_steps=args.warmup_steps),
         optimizer=args.optimizer,
         optimization_params={
             "num_epochs": args.num_epochs,
+            "max_steps": args.max_steps,
```

Comment:
What does it mean for the optimizer if a user specifies both num_epochs and max_steps?

Reply:
It stops depending on which one occurs first? Is this what we want?

Reply:
yes, it will stop based on whatever occurs first. I think this is a reasonable expectation